### ISSUE TITLE
Unable to use @nrwl/angular:component generator in custom nx workspace-generator

### ISSUE THREAD

**Issue by jrista** [nrwl/nx#10873]
## Current Behavior
When creating a simple custom workspace generator with Nx 13 (we are currently unable to use 14 due to the need to use Ionic, and the incompatibility of @nxtend/* with Nx 14), we are running into an issue where some of the generators, notably the angular component generator, are not properly included in the generators.d.ts file nor specified in the "exports" of the library's package.json file. 

We can import the necessary code, and implement the generator, however upon execution we encounter the following error:

```
Package subpath './src/generators/component/component' is not defined by "exports" in /Users/jon.rista/contracts/idnetworks/client-idnetworks-project/node_modules/@nrwl/angular/package.json
```

I am not sure why this is even an issue, or what is checking that this module is exported in this particular way. Compilation wise, importing it like this has not been an issue:

```ts
import { libraryGenerator } from '@nrwl/angular/generators';
import { componentGenerator } from '@nrwl/angular/src/generators/component/component';
```

However, it does not come from `@nrwl/angular/generators` like most of the others. We have implemented several custom workspace generators, several of which require the use of angular's component generator, and this issue has stopped us dead in our tracks.

## Expected Behavior

Angular component generator would be included in the exports for `@nrwl/angular/generators` allowing proper use of the component generator with it being correctly included in the public interface of the library. Compilation should succeed without error, and execution should run without error.

## Steps to Reproduce

Create a simple generator that uses the angular component generator. 

```
nx generate @nrwl/workspace:workspace-generator --name=angular-page --no-interactive
```

A simplified version of our generator is as follows:

```ts
import { libraryGenerator } from '@nrwl/angular/generators';
import { componentGenerator } from '@nrwl/angular/src/generators/component/component';
import { UnitTestRunner } from '@nrwl/angular/src/utils/test-runners';
import { formatFiles, installPackagesTask, Tree } from '@nrwl/devkit';
import { Linter } from '@nrwl/linter';

export interface Schema {
  name: string;
  tags?: string;
}

export default async function (tree: Tree, { name, group, tags }: Schema) {
  await libraryGenerator(tree, {
    name,
    directory: 'pages',
    prefix: 'app',
    routing: true,
    lazy: true,
    strict: true,
    compilationMode: 'full',
    linter: Linter.EsLint,
    unitTestRunner: UnitTestRunner.Jest,
    importPath: `~app/pages/${name}`,
    tags: `page ${tags ? `, ${tags}` : ''}`,
  });
  await componentGenerator(tree, {
    name,
    project: `pages-${name}`,
    style: 'scss',
    flat: true,
    skipSelector: true,
    type: 'page',
    path: `libs/pages/${name}/src/lib`,
  });

  await formatFiles(tree);
  return () => {
    installPackagesTask(tree);
  };
}
```

Schema is as follows:

```json
{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "angular-page",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Page name",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the page?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the library (used for linting)."
    }
  },
  "required": ["name", "group"]
}

```

To execute:

nx workspace-generator angular-page

### Failure Logs

```
What name would you like to use for the page? · test
Package subpath './src/generators/component/component' is not defined by "exports" in ~/app/node_modules/@nrwl/angular/package.json
```

### Environment

```
➜ nx report

 >  NX   Report complete - copy this into the issue template

   Node : 14.17.6
   OS   : darwin arm64
   npm  : 6.14.15
   
   nx : 13.10.5
   @nrwl/angular : 13.10.5
   @nrwl/cypress : 13.10.5
   @nrwl/detox : Not Found
   @nrwl/devkit : 13.10.5
   @nrwl/eslint-plugin-nx : 13.10.5
   @nrwl/express : Not Found
   @nrwl/jest : 13.10.5
   @nrwl/js : Not Found
   @nrwl/linter : 13.10.5
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 13.10.5
   @nrwl/web : Not Found
   @nrwl/workspace : 13.10.5
   typescript : 4.6.4
   rxjs : 7.4.0
   ---------------------------------------
   Community plugins:
         @ionic/angular: 5.9.4
         @ngrx/effects: 13.2.0
         @ngrx/store: 13.2.0
         @nxtend/capacitor: 13.0.0
         @nxtend/ionic-angular: 13.1.0

```

**Comment by jrista** [comment]
To be clear, merging the fix we required into Nx 14 does not actually solve the problem for our particular use case. I have provided details on the two PRs involved in fixing this issue, including the one that was merged to Nx 14, for more details.

It is my contention that this issue is not fixed in a way that will work for us, and should be re-opened given we are RESTRICTED to using Nx 13 for our use case, due to breaking changes in Nx 14 that broke @nxtend compatibility with that version. The @nxtend addons for Nx work up through Nx 13 to add Ionic and Capacitor support to the monorepo, they fail to work with any version of Nx 14 that we have tested. We spent over a week trying to get Nx 14 working, and finally settled on Nx 13 because the incompatibility of @nxtend/capacitor and @nxtend/ionic-angular with Nx 14. 

Since the fix was merged only to the Nx 14 mainline, and since we cannot use Nx 14, and since we have no timeline for which a fix for @nxtend to bring it into compatibility with Nx 14 may occur, we must use Nx 13 and as such the merged PR does not address the problem for us. I respectfully request that this issue be reopened until the issue can be resolved with the only version of Nx that is compatible with our particular use case. Thank you.
